### ISSUE TITLE
Question about executing PartA in Project1 (Can not output images successfully using CUDA)

### ISSUE THREAD

**Issue by LuoZhexu** [tonyyxliu/CUHKSZ-CSC4005#25]
I'm using Docker when executing these commands to test PartA in Project1.
![image](https://github.com/tonyyxliu/CSC4005-2023Fall/assets/115707762/d6b244cd-fdc8-4e0f-aaa8-9880a93d1e3d)
There exists CUDA on my host computer.
![image](https://github.com/tonyyxliu/CSC4005-2023Fall/assets/115707762/7da78fa8-186a-481c-9419-d30a2cbef42b)
**However when executing "./src/gpu/cuda_PartA /path/to/input.jpg /path/to/output.jpg" regarding with CUDA, my output image "my-Lena.jpg" and "my-4k.jpg" are all black. But when I execute with OpenACC, it works well as normal.**
![image](https://github.com/tonyyxliu/CSC4005-2023Fall/assets/115707762/cd22460b-46df-48d5-ab87-08e57b883565)
When I execute "nvidia-smi" command on the docker terminal, it also shows "no command found".
![image](https://github.com/tonyyxliu/CSC4005-2023Fall/assets/115707762/754c7e52-3f0b-40c9-802b-205a6eb3be59)

**Comment by tonyyxliu** [comment]
It sounds weird to me. Did you add the `--gpus all` flag when creating the docker container?

`docker run -it -d -p 2222:22 --gpus all --name csc4005 <IMAGE ID>`

**Comment by LuoZhexu** [comment]
I've followed all the instruction on "docs/docker_installation_guide.md" (I have NVIDIA on my host so I follow the corresponding part). But there's one reason that may cause the situation.
When I asked ustf about how to edit my code using docker, he creates another container "boring_wes..." as shown in the figure (which is the first container) and linked it with one of my host directory. So far what I've done is on this container. And the error I depicted before also happened on this container.
![image](https://github.com/tonyyxliu/CSC4005-2023Fall/assets/115707762/184b6812-7c03-4026-a9d5-acd41591a4a8)
The third container has just been created by me using the instruction you gave me recently. And when I enter the container, I execute nvidia-smi command and it can print the information normally.
![image](https://github.com/tonyyxliu/CSC4005-2023Fall/assets/115707762/eea094c9-4ab9-4fc1-877b-7032e4dfb7c7)
I don't know if the problem occurs when ustf replaced the container for me. And if so, how can I connect my current NVIDIA new container to some directory on my host?
